python/pathway/xpacks/llm/question_answering.py

```diff
@@ -563,6 +583,50 @@ class SummarizeQuerySchema(pw.Schema):
         self.StatisticsQuerySchema = self.indexer.StatisticsQuerySchema
         self.InputsQuerySchema = self.indexer.InputsQuerySchema
 
+    def _apply_reranking(self, pw_ai_results: pw.Table) -> pw.Table:
+        """Apply reranking to retrieved documents."""
+
+        @pw.udf
+        def add_score_to_doc(doc: pw.Json, score: float) -> dict:
+            return {**doc.as_dict(), "reranker_score": score}
+
+        # Flatten docs into rows
+        pw_ai_results_exploded = pw_ai_results.flatten(pw_ai_results.docs, origin_id='query_id')
+
+        # Apply reranker
+        pw_ai_results_scored = pw_ai_results_exploded.select(
+            pw.this.prompt,
+            pw.this.model,
+            pw.this.filters,
+            pw.this.return_context_docs,
+            pw.this.query_id,
+            doc=pw.this.docs,
+            reranker_score=self.reranker(pw.this.docs["text"], pw.this.prompt)
```

Comment:
```suggestion
            reranker_score=self.reranker(pw.this.docs["text"], pw.this.prompt),  # type: ignore
```
mypy fails to realize that reranker cannot be None here